airflow/providers/apache/kylin/operators/kylin.py

```diff
@@ -0,0 +1,166 @@
+    :type interval: int
+    :param timeout: timeout value,default value is 1 day,60 * 60 * 24 s
+    :type timeout: int
+    :param eager_error_status: jobs error status,if job status in this list ,this task will be error.
+        default value is tuple(["ERROR", "DISCARDED", "KILLED", "SUICIDAL", "STOPPED"])
+    :type eager_error_status: tuple
+    """
+    template_fields = ('project', 'cube', 'dsn', 'command', 'start_time', 'end_time',
+                       'segment_name', 'offset_start', 'offset_end')
+    ui_color = '#E79C46'
+
+    @apply_defaults
+    def __init__(self,
+                 kylin_conn_id: Optional[str] = 'kylin_default',
+                 project: Optional[str] = None,
+                 cube: Optional[str] = None,
+                 dsn: Optional[str] = None,
+                 command: Optional[str] = None,
+                 start_time: Optional[str] = None,
+                 end_time: Optional[str] = None,
+                 offset_start: Optional[str] = None,
+                 offset_end: Optional[str] = None,
+                 segment_name: Optional[str] = None,
+                 is_track_job: Optional[bool] = False,
+                 interval: int = 60,
+                 timeout: int = 60 * 60 * 24,
+                 eager_error_status=tuple(["ERROR", "DISCARDED", "KILLED", "SUICIDAL", "STOPPED"]),
+                 *args,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.kylin_conn_id = kylin_conn_id
+        self.project = project
+        self.cube = cube
+        self.dsn = dsn
+        self.command = command
+        self.start_time = start_time
+        self.end_time = end_time
+        self.segment_name = segment_name
+        self.offset_start = offset_start
+        self.offset_end = offset_end
+        self.is_track_job = is_track_job
+        self.interval = interval
+        self.timeout = timeout
+        self.eager_error_status = eager_error_status
+        self.jobs_error_status = [stat.upper() for stat in eager_error_status]
+        self.hook = None
+        self.cube_command = []
+        self.build_command = {'fullbuild', 'build', 'merge', 'refresh',
+                              'build_streaming', 'merge_streaming', 'refresh_streaming'}
+        self.jobs_end_status = ["FINISHED", "ERROR", "DISCARDED", "KILLED", "SUICIDAL", "STOPPED"]
+
+    def execute(self, context):
+
+        self.hook = KylinHook(kylin_conn_id=self.kylin_conn_id,
+                              project=self.project,
+                              dsn=self.dsn)
+
+        self.cube_command = self.hook.invoke_command
+        if self.command.lower() not in self.cube_command:
+            raise AirflowException('Kylin:Command {cmd} can not match kylin command list {cmds}'.format(
+                                   cmd=self.command, cmds=self.cube_command))
+
+        kylinpy_params = {
+            'start': datetime.fromtimestamp(int(self.start_time) / 1000) if self.start_time else None,
```

Comment:
Wouldn't it be simpler if the user could pass a datetime object or ISO datetime?

Reply:
In concert with the kylin API, and May be some users will use Jinja templates ,so millisecond timestamp  is more  appropriate（but kylinpy use datetime so it should be converted to datetime）